HealthCheck/Tools/class-health-check-tool.php

```diff
@@ -18,7 +13,11 @@ abstract class Health_Check_Tool {
 	protected $label;
 
 	public function __construct() {
-		add_filter( 'health_check_tools_tab', array( $this, 'tab_setup' ) );
+		// Legacy fallback. Please use init() in child classes for your bootstrap logic.
+	}
+
+	public function init() {
```

Comment:
This is the new `init()` bootstrap method that can be used by tools to register their business logic.

It is no longer coupled to the `__construct()` so the tool can be instantiated without actually registering the hooks which is useful during tests and lazy-loading complex business logic in general.